src/ynEIGEN/EigenStrategyManager.sol

```diff
@@ -301,34 +342,20 @@ contract EigenStrategyManager is
             IERC20 asset = assets[j];
             for (uint256 i; i < nodesCount; i++ ) {
                 ITokenStakingNode node = nodes[i];
-                
-                uint256 strategyBalance = toUserAssetAmount(
-                    asset,
-                    strategies[asset].userUnderlyingView((address(node)))
-                );
-                stakedBalances[j] += strategyBalance;
-            }
-        }
-    }
 
-    /**
-     * @notice Converts the user's underlying asset amount to the equivalent user asset amount.
-     * @dev This function handles the conversion for wrapped staked ETH (wstETH) and wrapped other ETH (woETH),
-     *      returning the equivalent amount in the respective wrapped token.
-     * @param asset The ERC20 token for which the conversion is being made.
-     * @param userUnderlyingView The amount of the underlying asset.
-     * @return The equivalent amount in the user asset denomination.
-     */
-    function toUserAssetAmount(IERC20 asset, uint256 userUnderlyingView) public view returns (uint256) {
-        if (address(asset) == address(wstETH)) {
-            // Adjust for wstETH using view method, converting stETH to wstETH
-            return wstETH.getWstETHByStETH(userUnderlyingView);
-        }
-        if (address(asset) == address(woETH)) { 
-            // Adjust for woETH using view method, converting oETH to woETH
-            return woETH.previewDeposit(userUnderlyingView);
+                uint256 strategyBalance = strategies[asset].userUnderlyingView((address(node)));
+
+                uint256 strategyWithdrawalQueueBalance;
+                (uint256 queuedShares, uint256 strategyWithdrawnBalance) = node.getQueuedSharesAndWithdrawn(strategies[asset], asset);
+                if (queuedShares > 0) {
+                    strategyWithdrawalQueueBalance = strategies[asset].sharesToUnderlyingView(queuedShares);
+                }
+
+                stakedBalances[j] +=
+                    wrapper.toUserAssetAmount(asset, strategyBalance + strategyWithdrawalQueueBalance) +
```

Comment:
for this optimization to be worthwhile, this contract call needs to be moved outside the inner loop.

Right now we still do 1 call per inner loop to `wrapper.toUserAssetAmount(asset, strategyBalance + strategyWithdrawalQueueBalance)`

To optimize this you need to rely on the property that

`wrapper.toUserAssetAmount(asset, X) + wrapper.toUserAssetAmount(asset, Y) == wrapper.toUserAssetAmount(asset, X + Y)`

Therefore you sum the uncoverted amounts inside the inner loop and convert the *sum* outside of the inner loop exactly once.

Reply:
done here https://github.com/yieldnest/yieldnest-protocol/pull/164/commits/be925af7b96e5449981372857a2287ed4cb63cb0